enumerable.rb

```diff
@@ -0,0 +1,134 @@
+# frozen_string_literal: true
+
+module Enumerable
+  def my_each
+    if block_given?
+      (0..length - 1).my_each do |i|
+        yield(self[i])
+      end
+    else
+      puts "You didn't pass in a block!"
+    end
+  end
+
+  def my_each_with_index
+    if block_given?
+      (0..length - 1).my_each do |i|
+        yield(self[i], i)
+      end
+    else
+      puts "You didn't pass in a block!"
+    end
+  end
```

Comment:
- We need to fix my_each first to make this method works as expected.

`['a','b','c'].each_with_index { |x,i| p x; p i }` should print:
```
"a"
0
"b"
1
"c"
2
```

- Just like the original each method. Your "my_each should return an enumerator if no block was passed.
`irb(main):002:0> [1,2,3,4,5].each_with_index`
`=> #<Enumerator: [1, 2, 3, 4, 5]:each_with_index>`
This can be achived returning `self.to_enum`